sql/core/src/main/scala/org/apache/spark/sql/SparkSQLParser.scala

```diff
@@ -74,9 +75,15 @@ private[sql] class SparkSQLParser(fallback: String => LogicalPlan) extends Abstr
     }
 
   private lazy val uncache: Parser[LogicalPlan] =
-    UNCACHE ~ TABLE ~> ident ^^ {
-      case tableName => UncacheTableCommand(tableName)
-    }
+    (
+      UNCACHE ~ TABLE ~> ident ^^ {
+        case tableName => UncacheTableCommand(tableName)
+      }
+    |
+      CLEAR ~ CACHE ^^ {
+        case _ ~ _ => ClearCacheCommand
+      }
+    )
```

Comment:
Nit: This can be simplified to

```scala
( UNCACHE ~ TABLE ~> ident ^^ {
    case tableName => UncacheTableCommand(tableName)
  }
| CLEAR ~ CACHE ^^^ ClearCacheCommand
)
```